ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -274,44 +287,55 @@ public void testContainerBuilder() throws JSONException {
       // build a container with arguments supplied
       containerBuilder =
           new ContainerBuilder(refContainerIds.get(i), refContainerNames.get(i), refContainerStatuses.get(i),
-              refContainerDescriptions.get(i), refContainerPrivacyValues.get(i), refAccountId);
+              refContainerDescriptions.get(i), refContainerEncryptionValues.get(i),
+              refContainerPreviousEncryptionValues.get(i), refContainerCachingValues.get(i),
+              refContainerMediaScanDisabledValues.get(i), refAccountId);
       Container containerFromBuilder = containerBuilder.build();
-      assertContainer(containerFromBuilder, i);
+      assertContainer(containerFromBuilder, i, false);
 
       // build a container from existing container
       containerBuilder = new ContainerBuilder(containerFromBuilder);
-      Container container2ByBuilder = containerBuilder.build();
-      assertContainer(container2ByBuilder, i);
+      containerFromBuilder = containerBuilder.build();
+      assertContainer(containerFromBuilder, i, false);
+
+      // test changing encryption setting through builder.
+      if (containerFromBuilder.isEncrypted()) {
+        // turn off and on, check that previouslyEncrypted is not set,
```

Comment:
let's make this a deterministic test

Reply:
generating the encryption settings in a deterministic way now